### DISCUSSION TITLE
CDT Call December 2024

### DISCUSSION THREAD

**Discussion by jonahgraham** [eclipse-cdt/cdt#958] [Announcements]
Our next CDT Call will be December 11th at 11am Ottawa time. Please make a comment to add suggestions for discussion topics, or come along to the call and we can discuss anything in "Any other business" section. Everyone is welcome.

## Call Information

<details><summary>Call Information and Agenda</summary>
<p>

- **Location:** [:computer: Zoom Meeting](https://eclipse.zoom.us/j/545364607?pwd=L1d2eDQ4RWJmdkk3YXkvcW90RWJiQT09) ([full details below](#Connection-Details))
- **Date:** December 11, 2024 11:00 AM Eastern Time (US and Canada) ([Calendar/ICS](https://eclipse.zoom.us/meeting/uJAodO-sqDstovAF50NQR9k2f8kNjAOvEg/ics?icsToken=98tyKuysqzgsHtKVt1z9c6ktOcH-b9_yln5Yi4UIzDHUMXlYMAWuPcxIE51pNd-B)) ([world clock](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2024&month=12&day=11&hour=16&min=0&sec=0&p1=188&p2=136&p3=195))
- **Agenda**
1. Welcome
2. Actions from last meeting
3. CDT 12.0 planning
  - CDT LSP default status / checkbox / etc
  - Core Build, especially for CMake
5. Any other business?
- **Contact:** <cdt-dev@eclipse.org> or [start a discussion topic](https://github.com/eclipse-cdt/cdt/discussions)
- **Reference:** [Previous and future meetings](https://github.com/eclipse-cdt/cdt/discussions?discussions_q=label%3Acdtcalls) (older call notes are located [here](https://hackmd.io/@jonahgraham/EclipseCDTCalls))

## Connection Details

Join Zoom Meeting
https://eclipse.zoom.us/j/545364607
Meeting ID: 545 364 607
Passcode: 196590
Dial by your location find your local number: https://eclipse.zoom.us/u/acMY3kyoJW

</p>
</details> 

## Minutes

- Agreement to remove new C Project and new C++ Project from the top level new project wizard. See https://github.com/eclipse-cdt/cdt/issues/965 for more details
  - A key condition of this removal is an ability for any ISV to restore it, possibly be adding some plugin.xml to their own bundles. This should work because the expectation to remove the wizards is simply removing some plugin.xml. 
- Clean up documentation about Downloads and auto-publish bleeding edge. See https://github.com/eclipse-cdt/cdt/issues/966
- CDT 12.0.0 will not be published as the latest (https://download.eclipse.org/tools/cdt/releases/latest/) build on release day to give time for early adopters and ISVs to test it. It is expected that the latest will be updated as part of the release process for SimRel 2025-03 in March
- Agreed on dates for 12.0.0 to align with 2025-03 milestones.
- Discussion on asciidoc work that is underway, See first page at https://github.com/eclipse-cdt/cdt/blob/main/doc/org.eclipse.cdt.doc.user/src/asciidoc/new_cmake_proj.adoc which ends up in the online and embedded eclipse help system too
- A description of Managed Build vs Core Build is being contributed to the help. See https://github.com/eclipse-cdt/cdt/pull/960
- Discussion on adding CDT LSP documentation to online help. See https://github.com/eclipse-cdt/cdt/issues/963

**Comment by ruspl-afed** [comment]
Unfortunately, I missed this call. 
I was interested to know about the intentions to improve the DSF API in order to copy less code during extension of GDB DSF: are they desirable at all or DSF API stability is more important.

**Reply by jonahgraham** [reply]
I assume this is in reference to #952? I don't have huge objection, if you find benefit in evolving the API go ahead. But keep in mind there is a very large install base that extends these services, so if we break API we have to be able to justify that to the community so they aren't annoyed at us. I would guess that if adding a new final method such as getAttributes without a major version bump, then it would be pretty ok. But I don't want to justify to all the existing extenders that we have broken the API without a tangible benefit to (at least some of) them.

**Reply by ruspl-afed** [reply]
> I assume this is in reference to https://github.com/eclipse-cdt/cdt/issues/952?

Yes, things like this. I'm about to finalize the experiment to inline GDB JTag code from Embedded CDT back to CDT and then see how compact Embedded CDT may be. The result is interesting, Embedded CDT may be much more compact (it may loose thousands LOC without loosing functionality). However I understand that there is no project today to accept such changes: Embedded CDT is focused on stability for wide range of CDT releases, while CDT DSF itself is also pretty fragile.

> if we break API 

No, I have no plans to introduce breaking changes, I would prefer to add new types and methods that would reduce the number of "levels" in the code and make its extension easier. 
But I need to ask one question: is DSF a kind of "deprecated" like MBS or not yet?

**Reply by jonahgraham** [reply]
>  is DSF a kind of "deprecated" like MBS or not yet?

Not from my perspective. It is complicated and very powerful, but there is nothing planned that could replace it.

But like all of CDT, it could use some love and attention.

**Comment by jonahgraham** [comment]
I updated the minutes to reflect what we discussed/agreed on at the meeting yesterday. See above. Much of it is links to issues/PRs that have been created since the meeting showing our action plan in full swing.
